privacyidea/lib/token.py

```diff
@@ -3007,19 +3009,32 @@ def import_tokens(tokens: list[dict], update_existing_tokens: bool = True, assig
         serial = token_info_dict.get("serial")
         try:
             existing_token = get_one_token(serial=serial, silent_fail=True)
-            if not existing_token:
-                token_type = token_info_dict.get("type")
-                db_token = Token(serial, tokentype=token_type.lower())
-                token = create_tokenclass_object(db_token)
-                token.import_token(token_info_dict)
-                successful_tokens.append(serial)
-                if assign_to_user and token_info_dict.get("user"):
-                    token.assign_user_during_token_import(token_info_dict.get("user"))
-            elif update_existing_tokens:
-                existing_token.import_token(token_info_dict)
+            # We check if there is no existing token or if we want to update existing tokens
+            if not existing_token or update_existing_tokens:
+
+                # We create a new token, if there is no existing token
+                if not existing_token:
+                    token_type = token_info_dict.get("type")
+                    db_token = Token(serial, tokentype=token_type.lower())
+                    db_token.save()
+                    token = create_tokenclass_object(db_token)
+                # We use the existing token and update it
+                else:
+                    token = existing_token
+
+                # Assign the user, if wanted and if there is a user in the token info dict
                 if assign_to_user and token_info_dict.get("user"):
-                    existing_token.assign_user_during_token_import(token_info_dict.get("user"))
-                updated_tokens.append(serial)
+                    owner = User(login=token_info_dict.get("user").get("login"),
+                                 resolver=token_info_dict.get("user").get("resolver"),
+                                 realm=token_info_dict.get("user").get("realm"),
+                                 uid=token_info_dict.get("user").get("uid"))
+                    token.add_user(owner, override=True)
+                token.import_token(token_info_dict)
+
+                if not existing_token:
+                    successful_tokens.append(serial)
+                else:
+                    updated_tokens.append(serial)
```

Comment:
The info about the parameter in the log message below can be removed. Just the info, that the token exists and maybe that it was not updated.